openmdao/visualization/connection_viewer/viewconns.py

```diff
@@ -206,41 +206,53 @@ def view_connections(root, outfile='connections.html', show_browser=True,
         'show_values': show_values,
     }
 
-    viewer = 'connect_table.html'
+    if '.html' in outfile:
+        viewer = 'connect_table.html'
 
-    code_dir = os.path.dirname(os.path.abspath(__file__))
-    libs_dir = os.path.join(os.path.dirname(code_dir), 'common', 'libs')
-    style_dir = os.path.join(os.path.dirname(code_dir), 'common', 'style')
+        code_dir = os.path.dirname(os.path.abspath(__file__))
+        libs_dir = os.path.join(os.path.dirname(code_dir), 'common', 'libs')
+        style_dir = os.path.join(os.path.dirname(code_dir), 'common', 'style')
 
-    with open(os.path.join(code_dir, viewer), "r", encoding='utf-8') as f:
-        template = f.read()
+        with open(os.path.join(code_dir, viewer), "r", encoding='utf-8') as f:
+            template = f.read()
 
-    with open(os.path.join(libs_dir, 'tabulator.5.4.4.min.js'), "r", encoding='utf-8') as f:
-        tabulator_src = f.read()
+        with open(os.path.join(libs_dir, 'tabulator.5.4.4.min.js'), "r", encoding='utf-8') as f:
+            tabulator_src = f.read()
 
-    with open(os.path.join(style_dir, 'tabulator.5.4.4.min.css'), "r", encoding='utf-8') as f:
-        tabulator_style = f.read()
+        with open(os.path.join(style_dir, 'tabulator.5.4.4.min.css'), "r", encoding='utf-8') as f:
+            tabulator_style = f.read()
 
-    jsontxt = json.dumps(data)
+        jsontxt = json.dumps(data)
 
-    with open(outfile, 'w', encoding='utf-8') as f:
-        s = template.replace("<connection_data>", jsontxt)
-        s = s.replace("<tabulator_src>", tabulator_src)
-        s = s.replace("<tabulator_style>", tabulator_style)
-        f.write(s)
-
-    if notebook:
-        # display in Jupyter Notebook
-        if not colab:
-            display(IFrame(src=outfile, width=1000, height=1000))
-        else:
-            display(HTML(outfile))
-
-    elif show_browser:
-        # open it up in the browser
-        from openmdao.utils.webview import webview
-        webview(outfile)
+        with open(outfile, 'w', encoding='utf-8') as f:
+            s = template.replace("<connection_data>", jsontxt)
+            s = s.replace("<tabulator_src>", tabulator_src)
+            s = s.replace("<tabulator_style>", tabulator_style)
+            f.write(s)
 
+        if notebook:
+            # display in Jupyter Notebook
+            if not colab:
+                display(IFrame(src=outfile, width=1000, height=1000))
+            else:
+                display(HTML(outfile))
+
+        elif show_browser:
+            # open it up in the browser
+            from openmdao.utils.webview import webview
+            webview(outfile)
+            
+    elif '.csv' in outfile:
```

Comment:
You should also add a test for the sake of code coverage. We don't test the  contents of the html file that is generated, but we do run it just to make sure there isn't an error: https://github.com/OpenMDAO/OpenMDAO/blob/777a589a88f0386d0d393ee0e5f028da1a7dfb64/openmdao/visualization/connection_viewer/tests/test_viewconns.py